### PR TITLE
[INFRA] Add .lgtm.yml file for better usage of LGTM CI tool

### LINKED ISSUE #864: [INFRA] LGTM alert
I tried to work around an LGTM alert with bb6065a in #853. Unfortunately, this doesn't work:
https://lgtm.com/projects/g/bids-standard/bids-specification/snapshot/56824aa6b869bcbec1fb14c50e3771cfc0358af5/files/pdf_build_src/process_markdowns.py?sort=name&dir=ASC&mode=heatmap#xea83c650d9db5491:1

I believe LGTM wants a lone `noqa` comment. If so, we need to change:
```python
from schemacode import macros # noqa (used in "eval" call later on)
```
into:
```python
# used in "eval" call later on
from schemacode import macros  # noqa
```
or, although I know you don't like it::
```python
# used in "eval" call later on
from schemacode import macros  # noqa: F417 lgtm [py/unused-import]
```

Last option, we could even revert to:
```python
# used in "eval" call later on
from schemacode import macros  # noqa: F417
```
and disable the relevant LGTM alert globally using an [`lgtm.yml` configuration file](https://lgtm.com/help/lgtm/lgtm.yml-configuration-file).

Which of these three options do you prefer? I prefer option 3, as LGTM is raising a false positive that cannot be worked around without sacrificing code quality (see https://github.com/github/codeql/issues/6517).

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Previous commit bb6065a from #853 does not work around the LGTM alert.

We attempt to silence the LGTM alert using an [`lgtm.yml`](https://lgtm.com/help/lgtm/lgtm.yml-configuration-file) file, and use standard Flake8 `noqa` suppression comments to document the issue at hand.

Fixes #864.

Links:
* https://lgtm.com/projects/g/bids-standard/bids-specification/snapshot/56824aa6b869bcbec1fb14c50e3771cfc0358af5/files/pdf_build_src/process_markdowns.py?sort=name&dir=ASC&mode=heatmap#xea83c650d9db5491:1
* https://github.com/github/codeql/issues/6517

### CODE REVIEW

**Review by sappelhoff — APPROVED** [review]
fine with me :-) the additional `lgtm.yml` file is a bit ugly in the repo, but I can live with it. Thanks!

**Comment by DimitriPapadopoulos** [comment]
The alternative to `lgtm.yml` would be to use plain `# noqa` instead of `#noqa: F401`. I feel it's a bad idea to lower code quality and precision to please a buggy tool.

**Comment by sappelhoff** [comment]
> The alternative to lgtm.yml would be to use plain # noqa instead of #noqa: F401. I feel it's a bad idea to lower code quality and precision to please a buggy tool.

I agree with you. the alternative that I see is to not care about the LGTM tool reports. Maybe someone else can offer their opinion here - can you tag some people you'd like to chime in?

**Comment by DimitriPapadopoulos** [comment]
The alternative is OK with me. It's just that a small part of the LGTM alerts is actually useful, so it would be useful to be able to disable (some of) the false positives and focus on real issues.

**Comment by sappelhoff** [comment]
@Remi-Gau what's your opinion? Add an `lgtm.yml` file to the repo in order to ignore false-positives from the [LGTM tool](https://lgtm.com/)  ... or ignore LGTM reports / live with the false positives?

I tend towards just merging the PR now.

**Comment by DimitriPapadopoulos** [comment]
That said, I'm not certain any more that `lgtm.yml` can disable the alerts displayed in LGTM.com:
https://lgtm.com/projects/g/bids-standard/bids-specification/

The `lgtm.yml` file in [pydicom/pydicom](https://github.com/pydicom/pydicom/blob/master/lgtm.yml) doesn't seem to be working:
https://lgtm.com/projects/g/pydicom/pydicom/

Will have to look into it in more detail.

**Comment by DimitriPapadopoulos** [comment]
Also, file can be renamed to a hidden file `.lgtm.yml`. Would you rather have a hidden file?

**Comment by Remi-Gau** [comment]
Ignoring false positive is the kind of implicit knowledge that might confuse new users / maintainers: it also leads experienced users / maintainers to not spot new true positive when they show up. So if we could avoid this it would be great, no?

**Comment by DimitriPapadopoulos** [comment]
The alternative would be to **document** the places where we suppress false positives. Like this:
```python
import foobar  # lgtm[py/unused-import]
```
However, that means adding multiple suppression comments for multiple linters:
```python
import foobar  # noqa: F401 # pylint: disable=unused-import # lgtm[py/unused-import] 
```

**Comment by DimitriPapadopoulos** [comment]
> Will have to look into it in more detail.

It actually does work. I was just confused by the existence of two alerts with very similar names, see https://github.com/pydicom/pydicom/pull/1508/commits/2ded1d2:
```
py/clear-text-logging-sensitive-data
py/clear-text-storage-sensitive-data
```

**Comment by sappelhoff** [comment]
Ok great, is it possible to include an explanatory comment in the `lgtm.yml` file that points to the tool (LGTM), the report, and the reason why this file is in the repo? Making the file hidden `.lgtm.yml` would be another welcome change.

I think after that we could merge, given Remi's opinion as well.

File: .lgtm.yml
Comment (sappelhoff) [inline .lgtm.yml:4]: in https://github.com/bids-standard/bids-specification/pull/865#issuecomment-918180786 I meant something like this:

```suggestion
---
# This file contains configuration for the LGTM tool: https://lgtm.com/
# The bids-specification repository is continuously scanned by the LGTM tool
# for any security and/or code vulnerabilities.
# You can find the report here: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
queries:
  # https://lgtm.com/rules/6770079/
  - exclude: py/unused-import
```

Where XXXXXXX is to be filled ;-)

Comment (DimitriPapadopoulos) [reply .lgtm.yml]: I'm not sure the link to the specific issue is a good idea. It's quite long:
https://lgtm.com/projects/g/bids-standard/bids-specification/snapshot/3d22fdf48aeb59b3203b3a31f291b375d1dbd17f/files/pdf_build_src/process_markdowns.py#x2d0047ad44974839:1

Comment (sappelhoff) [reply .lgtm.yml]: @DimitriPapadopoulos the documentation I propose is intended for  contributors/users who open this file and wonder what it is about: It's not a documentation about the specific queries. So I think XXXXXXXXXXXXX should be:  https://lgtm.com/projects/g/bids-standard/bids-specification/

Comment (DimitriPapadopoulos) [reply .lgtm.yml]: Ah, my wrong. I have been buried in these long URLs lately and had forgotten the base URL is very simple. Hope it's OK this time 😰

**Comment by sappelhoff** [comment]
Thanks @DimitriPapadopoulos

**Comment by DimitriPapadopoulos** [comment]
Because Semmle has joined GitHub, LGTM.com will be deprecated and replaced by GitHub code scanning.

[The next step for LGTM.com: GitHub code scanning!](https://github.blog/2022-08-15-the-next-step-for-lgtm-com-github-code-scanning/)

As far as I can understand, in simple cases such as this one, automated pull requests will be created to help us migrate:
> We will do our best to help migrate repositories that actively use LGTM.com to flag potential security issues in their pull requests. For those repositories, we will create pull requests that add a [GitHub Actions workflow that runs code scanning](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/setting-up-code-scanning-for-a-repository#setting-up-code-scanning-manually).

**Pull request merged**
